ping/ping.c

```diff
@@ -599,6 +636,10 @@ main(int argc, char **argv)
 		error(2, 0, "%s: %s", target, gai_strerror(ret_val));
 
 	for (ai = result; ai; ai = ai->ai_next) {
+		if (rts.opt_verbose)
+			printf("ai->ai_family: %s, ai->ai_canonname: %s\n",
+				   str_family(ai->ai_family), ai->ai_canonname);
```

Comment:
Possibly ai_canonname can be NULL too. I'm not sure what POSIX specifies on this and whether all the libc's that iputils are likely to be compiled with check for NULL.

Reply:
I haven't had a problem with `NULL`, but maybe I'm wrong. So would you change it to:
`, ai->ai_canonname ?: "");`